i3lock.c

```diff
@@ -138,6 +141,37 @@ static bool load_keymap(void) {
 }
 
 /*
+ * Loads the XKB compose table from the given locale.
+ *
+ */
+static bool load_compose_table(const char *locale) {
+    if (xkb_context == NULL) {
```

Comment:
You can delete line 148 to 153, xkb_context is guaranteed to be non-NULL at this point (it’s initialized in `load_keymap` and i3lock exits if that fails).